pkg/broker/provision_job.go

```diff
@@ -59,15 +59,17 @@ func (p *ProvisionJob) Run(token string, msgBuffer chan<- WorkMsg) {
 		return
 	}
 
-	msgBuffer <- ProvisionMsg{InstanceUUID: p.serviceInstance.ID.String(),
-		JobToken: token, SpecID: p.serviceInstance.Spec.ID, PodName: podName, Msg: "", Error: ""}
-
 	p.log.Info("Destroying APB sandbox...")
 	sm.DestroyApbSandbox(podName, p.serviceInstance.Context.Namespace)
 
 	// send creds
-	jsonmsg, _ := json.Marshal(extCreds)
-	p.log.Debug("sending message to channel")
+	jsonmsg, err := json.Marshal(extCreds)
+	if err != nil {
+		msgBuffer <- ProvisionMsg{InstanceUUID: p.serviceInstance.ID.String(),
+			JobToken: token, SpecID: p.serviceInstance.Spec.ID, PodName: "", Msg: "", Error: err.Error()}
+		return
```

Comment:
Hah I can't believe we were ignoring the error on Marshal. um this is definitely needed.